Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/AssetsManager.swift

```diff
@@ -0,0 +1,176 @@
+                ErrorHandler.handle(error: err)
+                throw err
+            }
+            
+            return UIImage(named: "\(cardNetwork.rawValue)-logo-colored", in: Bundle.primerResources, compatibleWith: nil)
+        }
+        
+        public static func getPaymentMethodAsset(for paymentMethodType: String) throws -> PrimerPaymentMethodAsset? {
+            if AppState.current.apiConfiguration == nil {
+                let err = PrimerError.uninitializedSDKSession(userInfo: nil, diagnosticsId: nil)
+                ErrorHandler.handle(error: err)
+                throw err
+            }
+            
+            guard let paymentMethod = PrimerAPIConfiguration.paymentMethodConfigs?.first(where: { $0.type == paymentMethodType }) else {
+                return nil
+            }
+            
+            guard let baseLogoImage = paymentMethod.baseLogoImage,
+                  let baseBackgroundColor = paymentMethod.displayMetadata?.button.backgroundColor
+            else {
+                return nil
+            }
+            
+            guard let paymentMethodLogo = PrimerPaymentMethodLogo(
+                colored: baseLogoImage.colored,
+                light: baseLogoImage.light,
+                dark: baseLogoImage.dark) else {
+                return nil
+            }
+            
+            guard let paymentMethodBackgroundColor = PrimerPaymentMethodBackgroundColor(
+                coloredStr: baseBackgroundColor.coloredHex,
+                lightStr: baseBackgroundColor.lightHex,
+                darkStr: baseBackgroundColor.darkHex) else {
+                return nil
+            }
+            
+            return PrimerPaymentMethodAsset(
+                paymentMethodType: paymentMethodType,
+                paymentMethodLogo: paymentMethodLogo,
+                paymentMethodBackgroundColor: paymentMethodBackgroundColor)
+        }
+        
+        public static func getPaymentMethodAssets() throws -> [PrimerPaymentMethodAsset] {
+            if AppState.current.apiConfiguration == nil {
+                let err = PrimerError.uninitializedSDKSession(userInfo: nil, diagnosticsId: nil)
+                ErrorHandler.handle(error: err)
+                throw err
+            }
+            
+            let hucAvailablePaymentMethods = PrimerHeadlessUniversalCheckout.PaymentMethod.availablePaymentMethods.compactMap({ $0.paymentMethodType })
+            
+            var paymentMethodAssets: [PrimerPaymentMethodAsset] = []
+            
+            for paymentMethod in (PrimerAPIConfiguration.paymentMethodConfigs ?? []) {
+                if !hucAvailablePaymentMethods.contains(paymentMethod.type) { continue }
+                
+                guard let baseLogoImage = paymentMethod.baseLogoImage,
+                      let baseBackgroundColor = paymentMethod.displayMetadata?.button.backgroundColor
+                else {
+                    continue
```

Comment:
So if we're missing just one data for a given payment method (for whatever reason), we don't return anything?